### ISSUE TITLE
Tuple array result in wrong encoding

### ISSUE THREAD

**Issue by mathewmeconry** [HashEx/abiencoder#21]
Given a parameters which is an array of tuple the encoding function produces a wrong result due to input restrictions.
The input field should enforce the user to enter something like this `[[1,2,3,4],[5,6,7,8]]` but it only takes in the values for one tuple.

This results in a wrong abi encoding for the function.

Inputs:
```
"0x11223344"
[["0x6b4584A05EB28016aDf0B0A692DD71073Fe4B593","0x123456",0]]
1234
56789
false
3
```

Generated ABI
`0xe910689700000000000000000000000000000000000000000000000000000000000000c0000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000004d2000000000000000000000000000000000000000000000000000000000000ddd500000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000003000000000000000000000000000000000000000000000000000000000000000411223344000000000000000000000000000000000000000000000000000000000000000000000000000000006b4584a05eb28016adf0b0a692dd71073fe4b5930000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000031234560000000000000000000000000000000000000000000000000000000000`

Expected ABI
`0xe910689700000000000000000000000000000000000000000000000000000000000000c0000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000004d2000000000000000000000000000000000000000000000000000000000000ddd50000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000300000000000000000000000000000000000000000000000000000000000000041122334400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000200000000000000000000000006b4584a05eb28016adf0b0a692dd71073fe4b5930000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000031234560000000000000000000000000000000000000000000000000000000000`

ABI
```json
[
  {
    "inputs": [
      {
        "internalType": "bytes",
        "name": "",
        "type": "bytes"
      },
      {
        "components": [
          {
            "internalType": "address",
            "name": "to",
            "type": "address"
          },
          {
            "internalType": "uint256",
            "name": "value",
            "type": "uint256"
          },
          {
            "internalType": "bytes",
            "name": "data",
            "type": "bytes"
          }
        ],
        "internalType": "struct IDAO.Action[]",
        "name": "",
        "type": "tuple[]"
      },
      {
        "internalType": "uint64",
        "name": "",
        "type": "uint64"
      },
      {
        "internalType": "uint64",
        "name": "",
        "type": "uint64"
      },
      {
        "internalType": "bool",
        "name": "",
        "type": "bool"
      },
      {
        "internalType": "enum IMajorityVoting.VoteOption",
        "name": "",
        "type": "uint8"
      }
    ],
    "name": "createVote",
    "outputs": [
      {
        "internalType": "uint256",
        "name": "voteId",
        "type": "uint256"
      }
    ],
    "stateMutability": "pure",
    "type": "function"
  }
]
```

**Comment by masquel** [comment]
@mathewmeconry thanks for the detailed example. 

Yes, we don't support encoding of `tuple[]` through web interface for now.

We have a plans to add it, and I hope we will be able to do it in a couple of days.

**Comment by masquel** [comment]
closed in #23 

Hi, @mathewmeconry 

We added support of `tuple[]` in our ABI encoder.

According to ABI you provided, your inputs should looks like:

```
0x11223344
[["0x6b4584A05EB28016aDf0B0A692DD71073Fe4B593",0,"0x123456"]]
1234
56789
false
3
```

We updated https://abi.hashex.org too.

Please, check it. Waiting for your feeeback
